lib/travis/services/find_log.rb

```diff
@@ -6,28 +6,26 @@ class FindLog < Base
       register :find_log
 
       def run(options = {})
+        return result_via_http if Travis.config.logs_api.enabled?
         result if result
       end
 
       def final?
-        # TODO jobs can be requeued, so finished jobs are no more final
-        # result && result.job && result.job.finished?
         false
       end
 
-      # def updated_at
-      #   result.updated_at
-      # end
-
-      private
-
-        def result
-          @result ||= if params[:id]
-            scope(:log).find_by_id(params[:id])
-          elsif params[:job_id]
-            scope(:log).where(job_id: params[:job_id]).first
-          end
+      private def result
+        if params[:id]
```

Comment:
below on line 26 you use a guard clause, but use the if/ifelse here, is there a reason for doing so? Are you looking to prevent a db call with a nil param?